cppcache/src/PdxHelper.cpp

```diff
@@ -53,10 +47,8 @@ PdxHelper::~PdxHelper() {}
 
 void PdxHelper::serializePdx(DataOutput& output,
                              const PdxSerializable& pdxObject) {
-  serializePdx(
-      output,
-      std::static_pointer_cast<PdxSerializable>(
-          std::const_pointer_cast<Serializable>(pdxObject.shared_from_this())));
+  serializePdx(output, std::const_pointer_cast<PdxSerializable>(
+                           pdxObject.shared_from_this()));
```

Comment:
Hmm.. something feels awkward about this.  Looks like we assume `pdxObject` is a smart pointer allocated on the heap w/control block.  If we want to increment ref count in this call would it make more sense to pass a shared_ptr as argument instead of explicitly calling `shared_from_this()`?

Reply:
Completely agree! I’d say we should pair up on GEODE-4359 and dig in